src/endpoint/s3/ops/s3_put_bucket_policy.js

```diff
@@ -10,19 +10,18 @@ const S3Error = require('../s3_errors').S3Error;
 async function put_bucket_policy(req) {
     let policy;
     try {
-        policy = JSON.parse(JSON.stringify(req.body).toLowerCase());
+        policy = JSON.parse(JSON.stringify(req.body));
     } catch (error) {
         console.error('put_bucket_policy: Invalid JSON provided', error);
         throw new S3Error(S3Error.InvalidArgument);
     }
-    // adapting bucket policy to our schema
-    for (const statement of policy.statement) {
-        if (statement.principal) {
-            if (statement.principal.aws) statement.principal = statement.principal.aws;
-            statement.principal = _.flatten([statement.principal]);
+    for (const statement of policy.Statement) {
+        if (statement.Principal) {
+            if (statement.Principal.AWS) statement.Principal = statement.Principal.AWS;
```

Comment:
Why do we still have this code to change the bucket policy? I thought the motivation for this change is that we just store the bucket policy as is.

Reply:
the reason I did it that way was because I didn't know schema can have more than one type for each field. I have changed the schema to support all types (object, array, string)